src/plugins/data/public/ui/dataset_selector/dataset_table.tsx

```diff
@@ -8,22 +8,12 @@ import { i18n } from '@osd/i18n';
 import { FormattedMessage } from '@osd/i18n/react';
 import React, { useRef, useState } from 'react';
 import { IDataPluginServices } from '../..';
-import { DataStructure } from '../../../common';
+import { DataStructureCreatorProps, DataStructure } from '../../../common';
 import { DatasetTypeConfig, DataStructureFetchOptions } from '../../query';
 import { getQueryService } from '../../services';
 
-interface DatasetTableProps {
+interface DatasetTableProps extends DataStructureCreatorProps<DataStructureFetchOptions> {
```

Comment:
nit:
since `DatasetTableProps` would it make more sense just make the base class like `DatasetCreationComponentProps` 

few reasons:
* we aren't really creating data structures we are trying to create a dataset
* `Component` because creator implies that the component itself will be the one responsible for creating.

Reply:
we can change name, i couldn't think of a good name. but

> we aren't really creating data structures we are trying to create a dataset

my understanding is that dataset is something user can search, but maybe this custom component would be used to create/select a data structure, which will have another custom component that would be used to create/select the dataset?